catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceInputTransformer.java

```diff
@@ -0,0 +1,333 @@
+import java.io.InputStream;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.boon.json.JsonFactory;
+import org.boon.json.ObjectMapper;
+import org.codice.ddf.confluence.common.Confluence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Media;
+import ddf.catalog.data.types.Security;
+import ddf.catalog.data.types.Topic;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+
+public class ConfluenceInputTransformer implements InputTransformer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfluenceInputTransformer.class);
+
+    private static final String PRODUCT_XML =
+            "<metadata><head><meta name=\"type\" content=\"%s\" /></head><body><p>%s</p></body></metadata>";
+
+    private static final String UNKNOWN = "UNKNOWN";
+
+    private static final String CONFLUENCE_TYPE_LINK =
+            "https://developer.atlassian.com/confdev/confluence-server-rest-api/advanced-searching-using-cql/cql-field-reference#CQLFieldReference-titleTitleType";
+
+    private static final int DESCRIPTION_SIZE = 256;
+
+    private static final ObjectMapper MAPPER = JsonFactory.create();
+
+    private MetacardType metacardType;
+
+    public ConfluenceInputTransformer(MetacardType type) {
+        metacardType = type;
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        Map<String, Object> json = getJsonObject(input);
+
+        return transformConfluenceResult(json, null, id);
+
+    }
+
+    public List<Metacard> transformConfluenceResponse(InputStream input)
+            throws IOException, CatalogTransformerException {
+
+        Map<String, Object> json = getJsonObject(input);
+        List<Metacard> metacards = new ArrayList<>();
+
+        String baseUrl = getString(json, "_links", "base");
+
+        getJsonArray(json, "results").stream()
+                .forEach(e -> {
+                    try {
+                        metacards.add(transformConfluenceResult(e, baseUrl, null));
+                    } catch (IOException | CatalogTransformerException ex) {
+                        LOGGER.error("Exception transforming confluence result.", ex);
```

Comment:
Since this is processing a list of results, I didn't want to have one bad result cause the rest not to be handled/returned.